src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -420,6 +420,9 @@ void saveProgram() throws IOException {
 
     @CpsVmThreadOnly
     public void saveProgram(File f) throws IOException {
+        boolean logging = LOGGER.isLoggable(Level.FINER);
```

Comment:
Anyway `isLoggable` is very cheap—negligible compared to the cost of the rest of the method. And checking this in a `static` initializer would block you from changing the logging level at runtime.